pkg/csi/service/driver.go

```diff
@@ -119,18 +120,33 @@ func (s *service) BeforeServe(
 	}
 
 	// Get the SP's operating mode.
-	s.mode = csictx.Getenv(ctx, gocsi.EnvVarMode)
-	if !strings.EqualFold(s.mode, "node") {
+	driver.mode = os.Getenv(csitypes.EnvVarMode)
+	if !strings.EqualFold(driver.mode, "node") {
 		// Controller service is needed
 		cfg, err = common.GetConfig(ctx)
 		if err != nil {
 			log.Errorf("failed to read config. Error: %+v", err)
 			return err
 		}
-		if err := s.cnscs.Init(cfg, Version); err != nil {
+		if err := driver.cnscs.Init(cfg, Version); err != nil {
 			log.Errorf("failed to init controller. Error: %+v", err)
 			return err
 		}
 	}
 	return nil
 }
+
+func (driver *vsphereCSIDriver) Run(ctx context.Context, endpoint string) {
```

Comment:
Can we add a comment for exported method `Run`.

Reply:
Done.